docs/manuals/gui/extension_file_structure.md

```diff
@@ -0,0 +1,92 @@
+# File Structure
+
+## Project Structure
+
+```
+- demo_lib
+  - frontend
+    - src --> this is where our react elements are stored
+      - index.ts --> this is where we export our taipy components
+    - webpack.config.js
+  - __init__.py
+  - demo_library.py
+- demo-init.py
+```
+- `demo_lib` is the directory where the whole extension library is located
+  - `demo_library.py` is the main library class that is used to define our elements
+  - `__init__.py` is used to shorten the import path
+  ```py
+  # __init__.py
+  from .demo_library import DemoLibrary
+  ```
+  - `frontend` is our react frontend code that is compiled into a bundle and served.
+    - `package.json`: JavaScript dependency file used by npm.<br/>
+          This file is updated when you manually install Taipy GUI.
+     - `webpack.config.js`: This file is used for building the JavaScript bundle that
+          holds the Web components code that is used in the generated pages.
+    - `tsconfig.json`: We are using [TypeScript](https://www.typescriptlang.org/)
+          as a more productive language, compared to JavaScript, to create the Web
+          components that are rendered on the generated pages. The TypeScript
+          transpiler (the program that transforms TypeScript code to vanilla JavaScript)
+          needs this file to drive its execution.
+    - `src` is where our react components are stored. Whenever you want to create new taipy component that is defined in `demo_library.py` you have to create it inside this folder
+    
+      - `index.ts` is where all react elements gets exported with the name that matches with our `demo_library.py` elements' name.
+    ```ts
+      // index.ts
```

Comment:
I am thinking whether we should keep a demo right here

Reply:
@dinhlongviolin1 Do you have any idea?